app/models/job_execution.rb

```diff
@@ -101,28 +101,18 @@ def execute!(dir)
 
     FileUtils.mkdir_p(artifact_cache_dir)
     @output.write("\n# Executing deploy\n")
-
-    commands = [
-      "export DEPLOYER=#{@job.user.email.shellescape}",
-      "export DEPLOYER_EMAIL=#{@job.user.email.shellescape}",
-      "export DEPLOYER_NAME=#{@job.user.name.shellescape}",
-      "export REVISION=#{@reference.shellescape}",
-      "export TAG=#{(@job.tag || @job.commit).to_s.shellescape}",
-      "export CACHE_DIR=#{artifact_cache_dir}",
-      "cd #{dir}",
-      *@job.commands
-    ]
+    @output.write("# Deploy URL: #{@job.deploy.url}\n") if @job.deploy
 
     payload = {
       stage: (stage.try(:name) || "none"),
       project: @job.project.name,
-      command: commands.join("\n")
+      command: commands(dir).join("\n")
```

Comment:
yeah ... it's cached but has an argument ... that smells like trouble :)
-> no cache but store as local variable